Source/FORMDataSource.m

```diff
@@ -513,36 +513,60 @@ - (void)fieldCell:(UICollectionViewCell *)fieldCell updatedWithField:(FORMField
             NSDictionary *parsed = [field.fieldID hyp_parseRelationship];
             NSString *sectionID = [NSString stringWithFormat:@"%@[%@]", [parsed objectForKey:@"relationship"], [parsed objectForKey:@"index"]];
             [self.formsManager sectionWithID:sectionID completion:^(FORMSection *section, NSArray *indexPaths) {
-                for (FORMSection *currentSection in section.form.sections) {
-                    if ([currentSection.position integerValue] > [section.position integerValue]) {
-                        currentSection.position = @([currentSection.position integerValue] - 1);
+                [self updateSectionPosition:section];
+
+                NSMutableArray *removedKeys = [NSMutableArray new];
+                __block NSDictionary *parsed;
+                [self.valuesDictionary enumerateKeysAndObjectsUsingBlock:^(NSString *key, id obj, BOOL *stop) {
+                    if ([key hasPrefix:section.sectionID]) {
+                        parsed = [key hyp_parseRelationship];
+                        NSString *newKey = [NSString stringWithFormat:@"%@[%@].%@", [parsed objectForKey:@"relationship"], @(self.formsManager.removedValues.count), [parsed objectForKey:@"attribute"]];
+                        [self.formsManager.values removeObjectForKey:key];
+                        [removedKeys addObject:newKey];
                     }
-                }
+                }];
+
+                parsed = [[section.sectionID stringByAppendingString:@".name"] hyp_parseRelationship];
```

Comment:
I just used `.name`, no real reference to anything. But `hyp_parseRelationship` wouldn't parse it without the suffix.